### PR TITLE
feat: 544 java gcs to mongo template

### PR DESCRIPTION
**Pull request by heyanand**
Creating pull request to merge changes for new java template gcstomongo

### CODE REVIEW

File: java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java [hunk 1]
```diff
@@ -375,22 +383,19 @@ public interface TemplateConstants {
 
   /** Common properties for all Kafka Based Templates * */
   String KAFKA_MESSAGE_FORMAT = "kafka.message.format";
-
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java:378]: Please run mvn spotless:apply

File: java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java:1]: Run `mvn spotless:apply`

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/README.md [hunk 2]
```diff
@@ -162,7 +161,48 @@ bin/start.sh \
 
 ```
 
-## 6. Text To BigQuery
+## 6. GCS To Mongo:
+
+Please download the MongoDb connector spark driver and copy it to GCS bucket location.
+
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/README.md:167]: Can we add some more info about the connector, probably the hyperlinks to maven repo?

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCStoMongo.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+          .option(MONGO_DATABASE, mongoDatabase)
+          .option(MONGO_COLLECTION, mongoCollection)
+          .option(MONGO_BATCH_SIZE, mongoBatchSize)
+          .mode(mongoSaveMode)
+          .save();
+    } catch (Throwable th) {
+      LOGGER.error("Exception in GCStoMongo", th);
+      if (Objects.nonNull(spark)) {
+        spark.stop();
+      }
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCStoMongo.java:107]: Incorrect usage of try-catch. Please remove try-catch.

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCStoMongo.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        default:
+          throw new IllegalArgumentException(
+              "Currently avro, parquet, csv and json are the only supported formats");
+      }
+      if (StringUtils.isAllBlank(mongoBatchSize)) {
+        mongoBatchSize = MONGO_DEFAULT_BATCH_SIZE;
+      }
+      if (StringUtils.isAllBlank(mongoSaveMode)) {
+        mongoSaveMode = SPARK_SAVE_MODE_APPEND;
+      }
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCStoMongo.java:93]: Can this be handled by template.properties file?

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -42,7 +43,7 @@ Please refer to the [Dataproc Templates (Java - Spark) README](/java)  for more
 * [S3ToBigQuery](/java/src/main/java/com/google/cloud/dataproc/templates/s3#1-s3-to-bigquery) (blogpost [link](https://medium.com/google-cloud/export-data-from-aws-s3-to-bigquery-using-dataproc-serverless-6dc7a9952fc4))
 * [SnowflakeToGCS](/java/src/main/java/com/google/cloud/dataproc/templates/snowflake#1-snowflake-to-gcs) (blogpost [link](https://medium.com/google-cloud/export-snowflake-query-results-to-gcs-using-dataproc-serverless-3d68f5a01ca9))
 * [SpannerToGCS](/java/src/main/java/com/google/cloud/dataproc/templates/databases#executing-spanner-to-gcs-template) (blogpost [link](https://medium.com/google-cloud/cloud-spanner-export-query-results-using-dataproc-serverless-6f2f65b583a4))
-* [TextToBigquery](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#6-text-to-bigquery)
+* [TextToBigquery](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#7-text-to-bigquery)
```
Comment (heyanand) [inline README.md:46]: Updated main readme

File: java/README.md [hunk 6]
```diff
@@ -32,7 +33,7 @@ Please refer to the [Dataproc Templates (Java - Spark) README](java/README.md)
 * [S3ToBigQuery](/java/src/main/java/com/google/cloud/dataproc/templates/s3#1-s3-to-bigquery)(blogpost [link](https://medium.com/google-cloud/export-data-from-aws-s3-to-bigquery-using-dataproc-serverless-6dc7a9952fc4))
 * [SnowflakeToGCS](/java/src/main/java/com/google/cloud/dataproc/templates/snowflake#1-snowflake-to-gcs)(blogpost [link](https://medium.com/google-cloud/export-snowflake-query-results-to-gcs-using-dataproc-serverless-3d68f5a01ca9))
 * [SpannerToGCS](/java/src/main/java/com/google/cloud/dataproc/templates/databases#executing-spanner-to-gcs-template) (blogpost [link](https://medium.com/google-cloud/cloud-spanner-export-query-results-using-dataproc-serverless-6f2f65b583a4))
-* [TextToBigquery](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#6-text-to-bigquery)
+* [TextToBigquery](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#7-text-to-bigquery)
```
[the remark this answers is not in the document]
Comment (heyanand) [reply java/README.md:36]: This is done

File: java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java
Comment (heyanand) [reply java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java:1]: Ran this and the file has been updated.

File: java/src/main/resources/template.properties [hunk 7]
```diff
@@ -268,6 +268,15 @@ gcs.gcs.write.mode=
...
 
+# GCS to Mongo
+gcs.mongodb.input.format=
+gcs.mongodb.input.location=
+gcs.mongodb.output.uri=
+gcs.mongodb.output.database=
+gcs.mongodb.output.collection=
+gcs.mongodb.batch.size=512
+gcs.mongodb.output.mode=append
+
```
Comment (heyanand) [inline java/src/main/resources/template.properties:279]: Updated template.properties to provide default values.

**Comment by heyanand** [comment]
@vanshaj-bhatia The integration tests are added in the Jenkinsfile and the tests passed without any issues. [Build#45] (https://dataproctemplatesci.com/job/dataproc-templates-build/job/ad-hoc/job/integration-tests-java/45/)

**Review by vanshaj-bhatia — APPROVED** [review]
LGTM

**Pull request merged**